ratis-server/src/main/java/org/apache/ratis/server/impl/RaftConfigurationImpl.java

```diff
@@ -252,7 +252,7 @@ boolean hasNoChange(Collection<RaftPeer> newMembers, Collection<RaftPeer> newLis
       if (inConf == null) {
         return false;
       }
-      if (!inConf.getRaftPeerProto().equals(peer.getRaftPeerProto())) {
+      if (inConf.getPriority() != peer.getPriority()) {
```

Comment:
@codings-dan , I guess comparing protos has caused test failures.  Why it is not working?  It seems comparing protos is correct since it will compare all the fields.

Reply:
I see now.  The hostnames could be different
- inConf: 
```
id: "s0"
address: "localhost:59232"
dataStreamAddress: "localhost:59235"
clientAddress: "localhost:59234"
adminAddress: "localhost:59233"
```
- peer  :
```
id: "s0"
address: "0.0.0.0:59232"
clientAddress: "0.0.0.0:59234"
adminAddress: "0.0.0.0:59233"
```

Reply:
@codings-dan, Let's compare only the priority as you have already done.  We should think about how to support updating hostnames and ports.  Perhaps it needs a new mode.

Reply:
yean, the hostname is different, I only change the priority.

Reply:
@szetszwo Yean, the hostname could be different, so I only compare the priority.

> We should think about how to support updating hostnames and ports

I will try, thanks!